akka-remote/src/main/scala/akka/remote/AddressUidExtension.scala

```diff
@@ -22,15 +24,24 @@ object AddressUidExtension extends ExtensionId[AddressUidExtension] with Extensi
   override def lookup = AddressUidExtension
 
   override def createExtension(system: ExtendedActorSystem): AddressUidExtension = new AddressUidExtension(system)
+
 }
 
 class AddressUidExtension(val system: ExtendedActorSystem) extends Extension {
+
   val longAddressUid: Long = {
-    // FIXME we should use a long here, but then we need to change in Cluster and RemoteWatcher also
-    //ThreadLocalRandom.current.nextLong()
-    ThreadLocalRandom.current.nextInt()
+    val tlr = ThreadLocalRandom.current
+    if (RARP(system).provider.remoteSettings.Artery.Enabled) tlr.nextLong()
```

Comment:
It's always a bit dangerous with extensions using other extensions from constructor, a cycle would deadlock.
RARP is just a glorified cast, so this should be fine.